pkg/crc/ssh/client.go

```diff
@@ -84,26 +87,49 @@ func (client *NativeClient) session() (*ssh.Session, error) {
 		if err != nil {
 			return nil, err
 		}
+
+		if err := client.conn.SetDeadline(time.Now().Add(timeout)); err != nil {
+			return nil, err
+		}
+		defer func() {
+			_ = client.conn.SetDeadline(time.Time{})
+		}()
+
 		c, chans, reqs, err := ssh.NewClientConn(client.conn, addr, config)
 		if err != nil {
 			return nil, err
 		}
 		client.sshClient = ssh.NewClient(c, chans, reqs)
 	}
+
+	if err := client.conn.SetDeadline(time.Now().Add(timeout)); err != nil {
+		return nil, err
+	}
+	defer func() {
+		_ = client.conn.SetDeadline(time.Time{})
+	}()
+
 	session, err := client.sshClient.NewSession()
 	if err != nil {
 		return nil, err
 	}
 	return session, err
 }
 
-func (client *NativeClient) Run(command string) ([]byte, []byte, error) {
-	session, err := client.session()
+func (client *NativeClient) RunWithTimeout(command string, timeout time.Duration) ([]byte, []byte, error) {
+	session, err := client.session(timeout)
 	if err != nil {
 		return nil, nil, err
 	}
 	defer session.Close()
 
+	if err := client.conn.SetDeadline(time.Now().Add(timeout)); err != nil {
+		return nil, nil, err
+	}
+	defer func() {
+		_ = client.conn.SetDeadline(time.Time{})
+	}()
+
```

Comment:
`client.session()` just did this, didn't it?

Reply:
> `client.session()` just did this, didn't it?

Ah no, `client.session()` did this, then it undid it, then we are doing it again :-/

Reply:
Yes the code is really ugly. I can't find a way to do it properly. I just added the block 

```
	if err := client.conn.SetDeadline(time.Now().Add(timeout)); err != nil {
		return nil, nil, err
	}
	defer func() {
		_ = client.conn.SetDeadline(time.Time{})
	}()
```

Before all interactions with the connection.

Reply:
If we add the SetDeadline only here, it wont timeout if in a middle of a Read or Write the user shutdowns the VM.

Reply:
> add a workaround for a podman issue which hopefully will be fixed when switching to OpenShift 4.8

Did that happen, or are we still waiting for an updated release?